### ISSUE TITLE
[HELP WANTED] Feature: Auto-load datasets from directory in SemanticModel

### ISSUE THREAD

**Issue by raphael-intugle** [Intugle/data-tools#136]
---
name: Help Wanted
about: Request community help on a specific task or feature
title: '[HELP WANTED] Feature: Auto-load datasets from directory in SemanticModel'
labels: 'help wanted'
assignees: ''
---

## What We Need Help With
We need to enhance the `SemanticModel` class to support initializing directly from a directory path. This feature will allow the system to automatically scan a specified folder for supported data files (CSV, Parquet, Excel) and load them as datasets into the semantic model using the DuckDB adapter.

## Background & Context
Currently, initializing a `SemanticModel` requires users to manually construct a dictionary or list of `DataSet` objects or configuration dictionaries. This creates unnecessary friction for users who simply want to point the tool at a local folder containing their data files and get started immediately. Simplifying this flow will improve the "getting started" experience.

## Current State
The `SemanticModel.__init__` method currently accepts `data_input` as either `Dict[str, Any]` or `List[DataSet]`. It does not support passing a file system path directly.

## Desired Outcome
A user should be able to initialize the model by passing a string path to a folder:
```python
sm = SemanticModel("./my_data_folder")
sm.build()
```
The system should automatically detect, configure, and load all supported files within that folder.

## Scope of Work
- [ ] Update `SemanticModel.__init__` signature and logic to accept a `str` type for `data_input`.
- [ ] Implement a private method `_initialize_from_folder(self, folder_path: str)` to handle the directory scanning.
- [ ] Implement logic to map file extensions to the appropriate DuckDB configuration `type`:
    - `.csv` -> `type="csv"`
    - `.parquet` -> `type="parquet"`
    - `.xlsx` -> `type="xlsx"`
- [ ] Instantiate `DataSet` objects for valid files and populate `self.datasets`.
- [ ] Add error handling for invalid paths or empty directories.

### Relevant Files/Modules
- `src/intugle/semantic_model.py`

### Key Concepts/Technologies
- **DuckDB Adapter**: This feature should rely on the existing `DuckdbAdapter` logic. You will need to construct configuration dictionaries (e.g., `{"path": "/abs/path/file.csv", "type": "csv"}`) that are compatible with it.
- **Python Pathlib/OS**: For robust file system traversal.

### Related Documentation
- See `src/intugle/adapters/types/duckdb/duckdb.py` for how DuckDB configurations are structured.

## Suggested Approach
1.  **Refactor `__init__`**: Modify the constructor in `src/intugle/semantic_model.py` to check if `data_input` is a string.
2.  **Validation**: Verify the string is a valid directory path.
3.  **Scanning**:
    - Iterate through files in the directory.
    - Use the file stem (filename without extension) as the dataset name.
    - Map the extension to the correct type (`csv`, `parquet`, `xlsx`).
4.  **Instantiation**:
    - Create the config dict: `config = {"path": absolute_file_path, "type": mapped_type}`.
    - Create the dataset: `dataset = DataSet(config, name=file_stem)`.
    - Add to `self.datasets`.

## Skills Needed
- [x] Python development
- [ ] Machine Learning (XGBoost, scikit-learn)
- [ ] LLM/GenAI integration (LangChain, OpenAI)
- [ ] Data engineering
- [ ] SQL and database knowledge
- [ ] Vector databases (Qdrant)
- [ ] Databricks/Snowflake experience
- [ ] Frontend/UI (Streamlit)
- [ ] Documentation writing
- [ ] Testing (pytest)

## Getting Started
1. Fork the repository.
2. Set up your development environment (see [CONTRIBUTING.md](https://github.com/Intugle/data-tools/blob/main/CONTRIBUTING.md)).
3. Locate `src/intugle/semantic_model.py`.
4. Implement the changes described in "Suggested Approach".

## Testing Requirements
- [x] Unit tests
    - Verify that passing a folder path correctly loads files.
    - Verify that unsupported files are ignored.
    - Verify correct mapping of extensions to types.
- [ ] Integration tests
- [ ] Manual testing with sample datasets
- [ ] Documentation examples

## Questions & Support
- Ask questions in the comments below
- Join our [Discord](https://discord.gg/NqR9tNWVTm) for real-time help
